frontend/src/Pages/Welcome.tsx

```diff
@@ -4,16 +4,61 @@ import { INIT_KRATOS_LOGIN_FLOW, ServerResponse, User } from '@/common';
 import API from '@/api/api';
 import { Link } from 'react-router-dom';
 import { AUTHCALLBACK } from '@/useAuth';
+import Timeline from '@/Components/Timeline';
+
+function useDomTheme(): 'light' | 'dark' {
+    const [theme, setTheme] = useState<'light' | 'dark'>(() => {
+        return (document.documentElement.getAttribute('data-theme') ??
+            'light') as 'light' | 'dark';
+    });
+
+    useEffect(() => {
+        const observer = new MutationObserver(() => {
+            const newTheme =
+                document.documentElement.getAttribute('data-theme') ?? 'light';
+            setTheme(newTheme as 'light' | 'dark');
+        });
+
+        observer.observe(document.documentElement, {
+            attributes: true,
+            attributeFilter: ['data-theme']
+        });
+
+        return () => {
+            observer.disconnect();
+        };
+    }, []);
+
+    return theme;
+}
 
 export default function Welcome() {
-    const [imgSrc, setImgSrc] = useState('unlockedv1Sm.webp');
+    const theme = document.documentElement.getAttribute('data-theme');
+    console.log(theme, '>>>>>>>>>'); // "light" or "dark"
+
     const [authUser, setAuthUser] = useState<User | undefined>();
+    const systemTheme = useDomTheme();
+
+    const [imgSrc, setImgSrc] = useState(() =>
+        systemTheme === 'light' ? 'unlockedv2DkSm.webp' : 'unlockedv2LtSm.webp'
+    );
+
+    useEffect(() => {
```

Comment:
This `useEffect` is also feels redundant since the next one is setting the image source as well. Though this one is setting it to `.webp` and the next one is `.png.` We should standartdize on one.